### PR TITLE
Psi4: Gracefully handle internal failures

### PR DESCRIPTION
**Pull request by dgasmith**
Psi4 internal failures are now handled gracefully. Also does not force `uks` reference when `reference` keyword is present.

### CODE REVIEW

File: qcengine/programs/psi4.py [hunk 1]
```diff
@@ -53,14 +53,16 @@ def compute(self, input_model: 'ResultInput', config: 'JobConfig') -> 'Result':
         if psi_version > self.parse_version("1.2"):
 
             mol = psi4.core.Molecule.from_schema(input_model)
-            if mol.multiplicity() != 1:
+            if (mol.multiplicity() != 1) and ("reference" not in input_model["keywords"]):
```
Comment (loriab) [inline qcengine/programs/psi4.py]: isn't `uks` a funny choice for generic, as opposed to `uhf`?

Comment (dgasmith) [reply qcengine/programs/psi4.py]: Yea, just hit that bug and realized why @ffangliu's example was failing.

Comment (dgasmith) [reply qcengine/programs/psi4.py]: See update.

Comment (ffangliu) [reply qcengine/programs/psi4.py]: Oh, I see. Thanks!

**Comment by dgasmith** [comment]
Looks like it cannot find Psi4? @loriab

**Comment by loriab** [comment]
failing in rdkit and torchani that don't have psi4 installed. needs something like `@using_psi4` above the test to only run when psi available.

**Pull request merged**